rclcpp_action/include/rclcpp_action/create_client.hpp

```diff
@@ -97,10 +97,10 @@ create_client(
  * \param[in] group The action client will be added to this callback group.
  *   If `nullptr`, then the action client is added to the default callback group.
  */
-template<typename ActionT>
+template<typename ActionT, typename NodeT = rclcpp::Node>
 typename Client<ActionT>::SharedPtr
 create_client(
-  rclcpp::Node::SharedPtr node,
+  typename NodeT::SharedPtr node,
```

Comment:
I think we should drop the shared pointer suffix. Then this function should also work with raw pointers.
```suggestion
  NodeT node,
```

IMO, an even better approach would be to implement what I proposed in https://github.com/ros2/rclcpp/pull/892#issuecomment-542465517, which requires implementing some missing free functions. This can be done in this PR or as a follow-up.

Same for `create_server`.

Reply:
Agreed, my goals are different than that of the ticket but I'm happy to give it a go to implement the free functions. I'm prefer if it were in another PR so that this isn't touch too many things & miss Friday's cutoff.

Looking over it, I think removing the `SharedPtr`would make sense if your prior comment on implicit deduction works. I would just be irritated as the user of this code if I had to specify the type of ptr each time I created a server & pretty much all nodes are going to be shared. If the deduction without specification doesn't work, I might push back to leave this as shared.

Reply:
Dropping `SharedPtr` seems to be working for me :slightly_smiling_face:

Reply:
nice, ok, will play around tomorrow